marshmallow/fields.py

```diff
@@ -931,62 +931,74 @@ class DateTime(Field):
 
     DEFAULT_FORMAT = 'iso'
 
+    OBJ_TYPE = 'datetime'
+
+    SCHEMA_OPTS_VAR_NAME = 'datetimeformat'
+
     localtime = False
     default_error_messages = {
-        'invalid': 'Not a valid datetime.',
-        'format': '"{input}" cannot be formatted as a datetime.',
+        'invalid': 'Not a valid {obj_type}.',
+        'format': '"{input}" cannot be formatted as a {obj_type}.',
     }
 
     def __init__(self, format=None, **kwargs):
         super(DateTime, self).__init__(**kwargs)
         # Allow this to be None. It may be set later in the ``_serialize``
         # or ``_desrialize`` methods This allows a Schema to dynamically set the
-        # dateformat, e.g. from a Meta option
-        self.dateformat = format
+        # format, e.g. from a Meta option
+        self.format = format
 
     def _add_to_schema(self, field_name, schema):
         super(DateTime, self)._add_to_schema(field_name, schema)
-        self.dateformat = self.dateformat or schema.opts.dateformat
+        self.format = self.format or \
+            getattr(schema.opts, self.SCHEMA_OPTS_VAR_NAME) or \
+            self.DEFAULT_FORMAT
 
     def _serialize(self, value, attr, obj):
+        self.format = self.format or self.DEFAULT_FORMAT
```

Comment:
This will conflict with https://github.com/marshmallow-code/marshmallow/pull/934.

Please don't mutate `self.format` and use a local `format` variable.

Reply:
Okay. It seems that change was merged just a few days ago. I'll go ahead and change `self.format` to just `format` in this case. Do you want me to change the name of the `self.format` variable back to `self.dateformat` as well? I changed it since `dateformat` in the `DateTime` field doesn't make sense.

Reply:
Sorry, I just realized this PR is rebased already. Then please change `self.format` to `format` as in #934.